docs/advanced-features/i18n-routing.md

```diff
@@ -204,6 +204,30 @@ export default function IndexPage(props) {
 }
 ```
 
+Note that when using [dynamic routes](/docs/routing/dynamic-routes) you will need to pass the `href` argument of `router.push` as an object.
```

Comment:
```suggestion
Note that when using [dynamic routes](/docs/routing/dynamic-routes) you might need to pass the `href` argument of `router.push` as an object when providing `as` manually as well.
```
This shouldn't be `required` when passing only the `href` for a dynamic route e.g. `router.push('/post/[slug]', '/post/hello-world',  { locale: 'fr' })` should work properly as well.